test-data/unit/lib-stub/dataclasses.pyi

```diff
@@ -0,0 +1,39 @@
+from typing import Any, Callable, Generic, Optional, TypeVar, overload
+
+_T = TypeVar('_T')
+_C = TypeVar('_C', bound=type)
+
+class InitVar(Generic[_T]):
+    ...
+
+
+@overload
+def dataclass(_cls: _C,
+              *,
+              init: bool = ...,
+              repr: bool = ...,
+              eq: bool = ...,
+              order: bool = ...,
+              unsafe_hash: bool = ...,
+              frozen: bool = ...) -> _C: ...
+
+
+@overload
+def dataclass(_cls: None = ...,
+              *,
+              init: bool = ...,
+              repr: bool = ...,
+              eq: bool = ...,
+              order: bool = ...,
+              unsafe_hash: bool = ...,
+              frozen: bool = ...) -> Callable[[_C], _C]: ...
+
+
+def field(*,
```

Comment:
It looks like current definition in typeshed is a bit different, could you please sync this one?

Reply:
This definition diverges from the one in typeshed in that it returns `_T` rather than `Field[_T]`. Returning `Field[_T]` breaks things like:

```
@dataclass
class Foo:
  x: int = field(default=42)
```

Because `int` and `Field[int]` are incompatible.